### PR TITLE
ft: update contract to new_syntax

### PR DESCRIPTION
**Pull request by codeWhizperer**

### CODE REVIEW

**Comment by enitrat** [comment]
cc @shramee

File: exercises/starknet/README.md [hunk 1]
```diff
@@ -1,26 +1,43 @@
 # Starknet Smart Contracts
 
-A Starknet contract can be created by annotating a Cairo module with the `#[contract]` attribute
+A Starknet contract can be created by annotating a Cairo module with the `#[starknet::contract]` attribute.
 
 Starknet contract don't require a `main` function.
-Functions in the contract module may be annotated as a `#[view]` or an `#[external]` function. External/View functions can be called by the users of Starknet, and by other contracts. The functions without these annotations are internal and cannot be accessed by users nor by other contract.
+Functions in the contract module may be annotated as `#[external(v0)]` function. External functions can be called by the users of Starknet, and by other contracts.
```
Comment (enitrat) [inline exercises/starknet/README.md:6]: > Functions in the contract module may be annotated as `#[external(v0)]` function.

Implementations can be annotated with `#[external(v0)]` and the functions defined inside them are accessible from outside the contract

File: exercises/starknet/README.md [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
+
+ The functions without these annotations are internal and cannot be accessed by users nor by other contract.
```
Comment (enitrat) [inline exercises/starknet/README.md:8]: > The functions without these annotations are internal and cannot be accessed by users nor by other contract.

The functions outside of these blocks are internal and cannot be accessed by users nor by other contract.

File: exercises/starknet/README.md [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
+
+ Functions in starknet contract have to explicitly define when writing to storage or reading from storage, by specifying the `ref self` when writing to storage or `self` when reading from state.
```
Comment (enitrat) [inline exercises/starknet/README.md:10]: ref self: ContractState
self: @ContractState

File: exercises/starknet/README.md [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+
+e.g 
+
+`fn write(ref self: ContractState){}`
+
+`fn read (self: ContractState) -> type{}`
```
Comment (enitrat) [inline exercises/starknet/README.md:16]: `fn read (self: @ContractState) -> T {}`

File: exercises/starknet/README.md [hunk 5]
```diff
@@ -1,26 +1,43 @@
...
+
+where `self` stands for the contract state, seeing the `self` pass by reference as arguement tell us that it mutates state, as it gives access to the contract storage. 
+
+
+#  Contract Interface
+Contract Interfaces is annotated with the `#[starknet::interface]` attribute. The constructor is not a part of the interface nor are internal function part of the interface.
```
Comment (enitrat) [inline exercises/starknet/README.md:22]: The contract interface is annotated

File: exercises/starknet/README.md [hunk 6]
```diff
@@ -1,26 +1,43 @@
...
 
 | Annotation       | Target     | Description                                     |
 | ---------------- | ---------- | ----------------------------------------------- |
-| `#[contract]`    | `module`   | Marks a module as a contract                    |
-| `#[constructor]` | `function` | Contract constructor, runs (only) on deployment |
-| `#[view]`        | `function` | An endpoint that doesn't modify state           |
-| `#[external]`    | `function` | An endpoint that modifies state                 |
+| `#[starknet::contract]`    | `module`   | Marks a module as a contract                    |
+| `#[constructor]` | `function` | Contract constructor, runs (only) on deployment |         |
+| `#[external(v0)]`    | `function` | An endpoint that modifies state
```
Comment (enitrat) [inline exercises/starknet/README.md:32]: `| `#[external(v0)]` | `impl` | Methods that modify state`

File: exercises/starknet/README.md [hunk 7]
```diff
@@ -84,5 +101,6 @@ fn play() {
 
 ## Further information
 
--   For more details, check out [The Structure of a Cairo Smart Contract](https://book.starknet.io/chapter_2/structure.html) in [the Starknet book](https://book.starknet.io).
+-   For more details, check out [A deep dive into Starknet Contract](https://book.cairo-lang.org/ch99-01-03-00-a-deeper-dive-into-contracts.html) in [the Cairo book](https://book.cairo-lang.org/).
```
Comment (enitrat) [inline exercises/starknet/README.md:104]: great :)

File: exercises/starknet/basics/starknet1.cairo [hunk 8]
```diff
@@ -1,22 +1,32 @@
...
+            // TODO
+        }
     }
 
 }
 
-#[abi]
-trait IJoesContract {
-    fn get_owner() -> felt252;
+#[starknet::interface]
```
Comment (enitrat) [inline exercises/starknet/basics/starknet1.cairo:26]: can you move the interface before the contract?

File: exercises/starknet/basics/starknet1.cairo [hunk 9]
```diff
@@ -1,22 +1,32 @@
...
 
-    fn get_owner() -> felt252 {
-        'Joe'
+#[storage]
+struct Storage{
+    owner:felt252
+}
+    #[external(v0)]
+    impl IJoesContract of super::IJoesContract<ContractState>{
+        fn set_owner(ref self:ContractState, _owner:felt252){
```
Comment (enitrat) [inline exercises/starknet/basics/starknet1.cairo:15]: remove `_` before owner parameter.
Also can you format the file please?

File: exercises/starknet/basics/starknet3.cairo [hunk 10]
```diff
@@ -7,30 +7,48 @@
 
 // I AM NOT DONE
 
-#[contract]
+use starknet::ContractAddress;
```
Comment (enitrat) [inline exercises/starknet/basics/starknet3.cairo:10]: insert empty line after

File: exercises/starknet/basics/starknet3.cairo [hunk 11]
```diff
@@ -7,30 +7,48 @@
...
     struct Storage {
         contract_owner: ContractAddress,
         // TODO: Set types for LegacyMap
         progress: LegacyMap<>
     }
 
     #[constructor]
-    fn constructor(owner: ContractAddress) {
-        contract_owner::write(owner);
+    fn constructor(ref self:ContractState,owner: ContractAddress) {
```
Comment (enitrat) [inline exercises/starknet/basics/starknet3.cairo:31]: format file

File: exercises/starknet/basics/starknet4.cairo [hunk 12]
```diff
@@ -1,38 +1,54 @@
...
 // starknet4.cairo
 // Liz, a friend of Jill, wants to manage inventory for her store on-chain.
 // This is a bit challenging for Joe and Jill, Liz prepared an outline
 // for how contract should work, can you help Jill and Joe write it?
 // Execute `starklings hint starknet4` or use the `hint` watch subcommand for a hint.
 
 // I AM NOT DONE
+use starknet::ContractAddress;
+
+#[starknet::interface]
```
Comment (enitrat) [inline exercises/starknet/basics/starknet4.cairo:11]: formatting

File: exercises/starknet/interoperability/starknet5.cairo [hunk 13]
```diff
@@ -27,57 +27,65 @@ mod ContractA {
...
+    #[storage]
     struct Storage {
         contract_b: ContractAddress,
         value: u128,
     }
 
     #[constructor]
-    fn constructor(_contract_b: ContractAddress) {
-        contract_b::write(_contract_b)
+    fn constructor(ref self: ContractState,_contract_b: ContractAddress) {
```
Comment (enitrat) [inline exercises/starknet/interoperability/starknet5.cairo:37]: remove `_` prefix

**Review by enitrat — COMMENTED** [review]
Great work!

**Comment by shramee** [comment]
Thanks for the review @enitrat 🙏 
And thanks for working on this @codeWhizperer 

I saw the first exercise seemed to be entirely different. Can we do this without changing the exercise itself?
If you want to propose changes to exercise itself. You can open an issue and we can proceed in a different PR.

For example the first exercise didn't use storage.

**Comment by codeWhizperer** [comment]
> Thanks for the review @enitrat pray And thanks for working on this @codeWhizperer
> 
> I saw the first exercise seemed to be entirely different. Can we do this without changing the exercise itself? If you want to propose changes to exercise, You can open an issue and we can proceed in a different PR.
> 
> For example the first exercise didn't use storage.

@shramee, I've created a new pull request and reverted the code in the first exercise to its previous state. Could you please review it? Thank you.

**Comment by enitrat** [comment]
@codeWhizperer you also need to update the Cairo dependencies to 2.1.0 in order to get Starklings to work. I will take care of it when I have time

**Comment by shramee** [comment]
Mergin this temporarily into [cairo-2.1.1](https://github.com/shramee/starklings-cairo1/tree/cairo-2.1.1 )

**Comment by codeWhizperer** [comment]
Okay ser, kindly let me know if there is any updates to be made to the
Repo.

**Pull request merged**